var/spack/repos/builtin/packages/win-wdk/package.py

```diff
@@ -0,0 +1,73 @@
+    version("10.0.16299", sha256="14efbcc849e5977417e962f1cd68357d21abf27393110b9d95983ad03fc22ef4", url="https://go.microsoft.com/fwlink/p/?linkid=859232", expand=False)
+    version("10.0.15063", sha256="489b497111bc791d9021b3573bfd93086a28b598c7325ab255e81c6f5d80a820", url="https://go.microsoft.com/fwlink/p/?LinkID=845980", expand=False)
+    version("10.0.14393", sha256="0bfb2ac9db446e0d98c29ef7341a8c8e8e7aa24bc72b00c5704a88b13f48b3cb", url="https://go.microsoft.com/fwlink/p/?LinkId=526733", expand=False)
+
+    # need one to one dep on SDK per https://github.com/MicrosoftDocs/windows-driver-docs/issues/1550
+    # additionally, the WDK needs to be paired with a version of the Windows SDK
+    # as per https://learn.microsoft.com/en-us/windows-hardware/drivers/download-the-wdk#download-icon-step-2-install-windows-11-version-22h2-sdk
+    depends_on("win-sdk@10.0.19041.0", when="@10.0.19041")
+    depends_on("win-sdk@10.0.18362.0", when="@10.0.18362")
+    depends_on("win-sdk@10.0.17763.0", when="@10.0.17763")
+    depends_on("win-sdk@10.0.17134.0", when="@10.0.17134")
+    depends_on("win-sdk@10.0.16299.0", when="@10.0.16299")
+    depends_on("win-sdk@10.0.15063.0", when="@10.0.15063")
+    depends_on("win-sdk@10.0.14393.0", when="@10.0.14393")
+
+    for plat in ["linux", "darwin", "cray"]:
+        conflicts("platform=%s" % plat)
+
+    @classmethod
+    def determine_version(cls, lib):
+        """
+        WDK is a set of drivers that we would like to
+        be discoverable externally by Spack.
+        The lib does not provide the WDK
+        version so we derive from the exe path
+        """
+        version_match_pat = re.compile(r"[0-9][0-9].[0-9]+.[0-9][0-9][0-9][0-9][0-9]")
+        ver_str = re.search(version_match_pat, lib)
+        return ver_str if not ver_str else Version(ver_str.group())
+
+    def setup_dependent_environment(self):
+        # This points to all core build extensions needed to build
+        # drivers on Windows
+        os.environ["WDKContentRoot"] = self.prefix
+
+    @run_before("install")
+    def rename_downloaded_executable(self):
+        """WGL download is named by fetch based on name derived from Link redirection
+        This name is not properly formated so that Windows understands it as an executable
+        We rename so as to allow Windows to run the WGL installer"""
+        installer = glob.glob(os.path.join(self.stage.source_path, "linkid=**"))
+        if len(installer) > 1:
+            raise RuntimeError("Fetch has failed, unable to determine installer path")
+        installer = installer[0]
+        os.rename(installer, os.path.join(self.stage.source_path, "wdksetup.exe"))
+
+    def install(self, spec, prefix):
+        install_args = ["/features", "+", "/quiet", "/installpath", self.prefix]
+        with working_dir(self.stage.source_path):
+            Executable("wdksetup.exe")(*install_args)
```

Comment:
You mentioned that users can't install `wdk` if it's already on the system: could the error message be augmented with a suggestion to run `spack external find --not-buildable wdk`?

Reply:
Will do

Reply:
I tested this, and the installer will report that scenario to the users prior to closure without Spack needing to do so. When spack runs `Executable("wdksetup.exe")(*install_args)` execution is handed over to the Windows installer and gui, which then detects other installations of the WDK and reports to that effect. Context is not returned to spack until a user acknowledges that scenario, so I don't believe reporting that would be necessary. Additionally, I don't know if the installer returns enough info to Spack via stdout that we would be able to differentiate that failure from another.

Reply:
Yes, we would be able to search for the registry key and report the already installed version. Windows installer APIs are not consistent across packages, each package creates their own. The WDK does not have the `-unattended` flag, instead the `-quiet` flag that I've already added handled both the normal interface and any error messages that may crop up. There are two to dos here, the first is the registry stuff, and the second is piping errors to the spack log file. I'll try to do both in this PR.

Reply:
If you point the log at spack's log it blows things away, that will be saved for a later PR.